pkg/chunk/gcp/storage_client.go

```diff
@@ -211,22 +295,22 @@ func (s *storageClientColumnKey) QueryPages(ctx context.Context, query chunk.Ind
 
 // bigtableReadBatchColumnKey represents a batch of values read from Bigtable.
 type bigtableReadBatchColumnKey struct {
+	i            int
 	items        []bigtable.ReadItem
 	columnPrefix string
 }
 
-func (b bigtableReadBatchColumnKey) Len() int {
-	return len(b.items)
+func (b *bigtableReadBatchColumnKey) Next() bool {
+	b.i++
+	return b.i < len(b.items)
 }
 
-func (b bigtableReadBatchColumnKey) RangeValue(index int) []byte {
-	return []byte(
-		strings.TrimPrefix(b.items[index].Column, b.columnPrefix),
-	)
+func (b *bigtableReadBatchColumnKey) RangeValue() []byte {
+	return []byte(strings.TrimPrefix(b.items[b.i].Column, b.columnPrefix))
```

Comment:
Consider inlining the columnPrefix since it is static? Then it would not need to be on `bigtableReadBatchColumnKey`